Cargo.toml

```diff
@@ -10,4 +10,9 @@ description = "A sha1 implementation for use in const contexts"
 keywords = ["const", "sha1", "hash", "digest"]
 readme = "README.md"
 
-[dependencies]
\ No newline at end of file
+[dependencies]
+
+
+[features]
+default = ["std"]
+std = []
```

Comment:
Personally, I think it's better to be explicit about `no_std` support, despite no std specific function is used other than in tests.

With this feature, we can add `#![cfg_attr(not(feature = "std"), no_std)]`. This will force the compiler not to link `std` and can help prevent regression issues.  Maybe should be added to CI as well?

Reply:
Also, this feature can be used to enable certain optimisation which is only available in std environment.